packages/core/src/snapshot.js

```diff
@@ -86,8 +102,10 @@ function mapSnapshotOptions(snapshots, context) {
     // transform snapshot URL shorthand into an object
     if (typeof snapshot === 'string') snapshot = { url: snapshot };
 
-    // normalize the snapshot url and use it for the default name
+    validateAndFixSnapshotUrl(snapshot);
```

Comment:
Why not calling this method in `validURL` method, single place for all URL validation

Reply:
- valid URL is used to create URL obj, and has been used in other places as well in snapshot.js, like getting allowedHostname, so do not want to modify any such URL at any place apart from `snapshot.url`, so created a separate function for that.

Reply:
All the URL validation should have encoding ideally, and should return URL object with proper encoding applied, if I need to use validURL and want encoding as well I will need to call two methods. Either take it as argument if encoding is required or not so that in future I can just pass encode option in this method and it will return correct URL

Reply:
- Will be testing `npx percy snapshot command` + defer uploads as well.